src/applications/yellow-ribbon/actions/index.js

```diff
@@ -102,15 +85,18 @@ export const fetchResultsThunk = (options = {}) => async dispatch => {
     });
 
     // Track the API request.
-    recordEvent({
-      event: 'edu-yellow-ribbon-search',
-      'edu-yellow-ribbon-name': name || undefined,
-      'edu-yellow-ribbon-state': state || undefined,
-      'edu-yellow-ribbon-city': city || undefined,
-      'edu-yellow-ribbon-contribution-amount': contributionAmount || undefined,
-      'edu-yellow-ribbon-number-of-students': numberOfStudents || undefined,
-      'edu-yellow-ribbon-number-of-search-results': response?.totalResults,
-    });
+    if (trackSearch) {
+      recordEvent({
+        event: 'edu-yellow-ribbon-search',
+        'edu-yellow-ribbon-name': name || undefined,
+        'edu-yellow-ribbon-state': state || undefined,
+        'edu-yellow-ribbon-city': city || undefined,
+        'edu-yellow-ribbon-contribution-amount':
+          contributionAmount || undefined,
+        'edu-yellow-ribbon-number-of-students': numberOfStudents || undefined,
+        'edu-yellow-ribbon-number-of-search-results': response?.totalResults,
+      });
+    }
```

Comment:
Hm, I suppose instead of a `trackSearch` flag, you could `recordEvent` from the reducer by first checking state to see if it's the first API call or not. Totally your call though. I'm curious about `recordEvent` in the reducer but I don't think that's something the project usually does otherwise.

Reply:
Ya I think I'll leave it as is since I'd prefer the side-effects to be all in one place (e.g. the thunk)

Reply:
Oh yeah, good point about side effects!